llvm/lib/CodeGen/GlobalISel/LegalizerHelper.cpp

```diff
@@ -3330,8 +3330,12 @@ LegalizerHelper::LegalizeResult LegalizerHelper::lowerLoad(GAnyLoad &LoadMI) {
   LLT MemTy = MMO.getMemoryType();
   MachineFunction &MF = MIRBuilder.getMF();
 
-  unsigned MemSizeInBits = MemTy.getSizeInBits();
-  unsigned MemStoreSizeInBits = 8 * MemTy.getSizeInBytes();
+  unsigned MemSizeInBits = MemTy.isScalable()
+                               ? MemTy.getSizeInBits().getKnownMinValue()
+                               : MemTy.getSizeInBits();
```

Comment:
I'm not sure that you need to do this. Can't you say `unsigned MemSizeInBits = MemTy.getSizeInBits().getKnownMinValue()`? When the vector is fixed `getKnownMinValue` is implicitly called on `MemTy.getSizeInBits()` to convert to unsigned.